packages/database/lib/migrations/20250723214519_create_pre_provisioned_providers.cjs

```diff
@@ -0,0 +1,20 @@
+/**
+ * @param {import('knex').Knex} knex
+ */
+exports.up = async function (knex) {
+    return knex.schema.createTable('_nango_pre_provisioned_providers', function (table) {
+        table.increments('id').primary();
+        table.string('name').notNullable();
+        table.string('oauth_client_id').nullable().defaultTo(null);
+        table.string('oauth_client_secret').nullable().defaultTo(null);
+        table.string('oauth_scopes').nullable().defaultTo(null);
```

Comment:
I would suggest using json column and using the interface we know, so we are not limited and we add some weird ones 
```ts
credentials: OAuth1Credentials | OAuth2ClientCredentials
```